shadows/framework/src/main/java/org/robolectric/shadows/ShadowCryptoObject.java

```diff
@@ -0,0 +1,60 @@
+package org.robolectric.shadows;
+
+import static android.os.Build.VERSION_CODES.P;
+import static android.os.Build.VERSION_CODES.R;
+import static android.os.Build.VERSION_CODES.TIRAMISU;
+
+import android.hardware.biometrics.CryptoObject;
+import android.security.identity.IdentityCredential;
+import android.security.identity.PresentationSession;
+import java.security.Signature;
+import javax.crypto.Cipher;
+import javax.crypto.Mac;
+import org.robolectric.annotation.Implementation;
+import org.robolectric.annotation.Implements;
+
+@SuppressWarnings("UnusedDeclaration")
+@Implements(value = CryptoObject.class, isInAndroidSdk = false, minSdk = P)
+public class ShadowCryptoObject {
+
+  @Implementation(minSdk = P)
+  public void __constructor__(Signature signature) {}
```

Comment:
Why shadow all of these constructors? They seem to be pure Java code:
https://cs.android.com/android/platform/superproject/+/master:frameworks/base/core/java/android/hardware/biometrics/CryptoObject.java;l=39?q=cryptoobject.java

I generally advise against shadowing constructors as it prevents initialization of the object, which can often lead to NPEs.

Reply:
See the stacktrace here: https://github.com/robolectric/robolectric/pull/8250#issuecomment-1573298274

Notice  `CryptoObject.getOpId()` which is **hidden**.
This method invokes `AndroidKeyStoreProvider.getKeyStoreOperationHandle()` based on the value of `mObject`. If `mObject` is null, then we have no problem. But all constructors are non-null so this is the only way to keep it null.

Reply:
If getOpId is the issue, why not just shadow that? Robolectric can shadow private/protected/`@hide` methods.

Reply:
Okay, I was not aware that this was possible. Also I don't know how it should work on different API levels as the method could have been removed/added at an undisclosed version. 

I'll have a look at other shadows when I have the time to see if there is any examples of this

Reply:
I've now updated the shadow to only override the problematic getOpId() function.

I've verified the fix locally.